lib/task/import/csvImportTask.class.php

```diff
@@ -1198,6 +1110,82 @@ public function execute($arguments = array(), $options = array())
       $ret = $buildNestedSet->run();
     }
   }
+
+  /**
+   * Handle creating a digital object and linking it to our information object during import.
+   *
+   * @param QubitFlatfileImport $self  A reference to our flat file importer ($self->object refers to
+   *                                   the information object we're currently creating).
+   * @param string $path  Asset file path
+   */
+  private function handleDigitalObjectPath($self, $path)
+  {
+    if (!is_readable($path))
+    {
+      $this->log("Cannot read digital object path. Skipping creation of digital object ($path)");
+      return;
+    }
+
+    $do = new QubitDigitalObject;
+    $do->usageId = QubitTerm::MASTER_ID;
+    $do->informationObject = $self->object;
+
+    // Don't create derivatives (reference, thumb)
+    if ($self->status['options']['skip-derivatives'])
+    {
+      $do->createDerivatives = false;
+    }
+
+    $do->assets[] = new QubitAsset($path);
+
+    try
+    {
+      $do->save();
+    }
+    catch (Exception $e)
+    {
+      $this->log($e->getMessage(), sfLogger::ERR);
+    }
+  }
+
+  /**
+  * Return default parent id based on various CLI options.
+  *
+  * @param string $sourceName  The source name of this file
+  * @param array $options  CLI options
+  * @return mixed  The default parent id
+  */
+  private function getDefaultParentId($sourceName, $options)
+  {
+    // Allow default parent ID to be overridden by CLI options
+    if ($options['default-parent-slug'])
+    {
+      $parentId = QubitFlatfileImport::getIdCorrespondingToSlug($options['default-parent-slug']);
+
+      if (!$options['quiet'])
+      {
+        $this->log("Parent ID of slug {$options['default-parent-slug']} is $parentId");
+      }
+    }
+    else if ($options['default-legacy-parent-id'])
+    {
+      if (false === $keyMapEntry = QubitFlatfileImport::fetchKeymapEntryBySourceAndTargetName(
+          $options['default-legacy-parent-id'], $sourceName, 'information_object'))
+      {
+        throw new sfException('Could not find keymap entry for default legacy parent ID '.
+                              $options['default-legacy-parent-id']);
+      }
+
+      $parentId = $keyMapEntry->target_id;
+      $this->log("Using default parent ID $parentId (legacy parent ID {$options['default-legacy-parent-id']})");
```

Comment:
Using sprintf instead of the curly braces to embed the option value might be cleaner, but that's subjective I guess so feel free to disregard.